tests/operator_test.go

```diff
@@ -1720,9 +1721,18 @@ spec:
 			By("waiting until the owner reference disappears again")
 			Eventually(func() []metav1.OwnerReference {
 				crd, err = virtClient.ExtensionsClient().ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), "virtualmachineinstances.kubevirt.io", metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
+				if err != nil {
+					if errors.IsNotFound(err) {
+						// the garbage collector might have deleted the object while Waiting
+						// for the update. Return original ownerRef so the test does not pass
+						return ownerRef
+					}
```

Comment:
@ashleyschuett

I'm unsure how this is possible for the CRD to not exist. The CRD shouldn't be deleted or garbage collected during the update.

Is the CRD being deleted due to the owner reference to a non-existent object?  If that's the case, maybe we can create a real resource (like a config map) and use a real object as the owner reference so garbage collection won't occur.

Reply:
> Is the CRD being deleted due to the owner reference to a non-existent object? If that's the case, maybe we can create a real resource (like a config map) and use a real object as the owner reference so garbage collection won't occur.

Yup, its collected because the owner reference does not exist. I have updated this test to instead reference a resource that does exist. Thanks!